src/ng/directive/ngModel.js

```diff
@@ -878,6 +878,141 @@ NgModelController.prototype = {
    */
   $overrideModelOptions: function(options) {
     this.$options = this.$options.createChild(options);
+  },
+
+  /**
+   * @ngdoc method
+   *
+   * @name  ngModel.NgModelController#$processModelValue
+
+   * @description
+   *
+   * Runs the model -> view pipeline on the current
+   * {@link ngModel.NgModelController#$modelValue $modelValue}
+   *
+   * The following actions are performed by this method:
+   *
+   * - the $modelValue is run through the {@link ngModel.NgModelController#$formatters $formatters}
+   * and the result is set to the {@link ngModel.NgModelController#$viewValue $viewValue}
+   * - the `ng-empty` or `ng-not-empty` class is set on the element
+   * - if the $viewValue has changed
+   *   - {@link ngModel.NgModelController#$render $render} is called on the control
+   *   - the {@link ngModel.NgModelController#$validators $validators} are run and
+   *   the validation status is set.
+   *
+   * This method is called by ngModel internally when the bound scope value changes.
+   * Application developers usually do not have to call this function themselves.
+   *
+   * This function can be used when the $viewValue or the rendered DOM value of the control should
+   * be updated after a user input.
+   *
+   * For example, consider a text input with an autocomplete list (for fruit, where the items are
+   * objects with a name and an id.
+   * A user enters `ap` and then selects `Apricot` from the list.
+   * Based on this, the autocomplete widget will call $setViewValue({name: 'Apricot', id: 443}),
+   * but the rendered value will still be `ap`.
+   * The widget can then call ctrl.$processModelValue() to run the model -> view
+   * pipeline again, which includes a formatter that converts the object into the string `Apricot`
+   * which is set to the $viewValue and rendered in the DOM.
+   *
+   * @example
+   * <example module="inputExample" name="ng-model-process">
+     <file name="app.js">
+      angular.module('inputExample', [])
+        .directive('processModel', function() {
+          return {
+            require: 'ngModel',
+            link: function(scope, element, attrs, ngModel) {
+
+              ngModel.$formatters.push(function(value) {
+                if (angular.isObject(value) && value.name) {
+                  return value.name;
+                }
+
+                return value;
+              });
+
+              ngModel.$parsers.push(function(value) {
+                if (angular.isString(value)) {
+                  return scope.items.find(function(item) {
```

Comment:
This is not ES5-compatible :grin:

Reply:
I don't feel too strongly, but it would be good for the docs to show code that works in all supportd browsers. It would also be a first (afaik).